lib/archethic/oracle_chain/scheduler.ex

```diff
@@ -140,21 +140,58 @@ defmodule Archethic.OracleChain.Scheduler do
 
     index = Map.fetch!(indexes, summary_date)
 
-    if trigger_node?(summary_date, index + 1) do
-      new_oracle_data =
-        summary_date
-        |> Crypto.derive_oracle_address(index)
-        |> get_oracle_data()
-        |> Services.fetch_new_data()
+    new_oracle_data =
+      summary_date
+      |> Crypto.derive_oracle_address(index)
+      |> get_oracle_data()
+      |> Services.fetch_new_data()
+
+    authorized_nodes =
+      summary_date
+      |> P2P.authorized_nodes()
+      |> Enum.filter(& &1.available?)
+
+    storage_nodes =
+      summary_date
+      |> Crypto.derive_oracle_address(index)
+      |> Election.storage_nodes(authorized_nodes)
+
+    tx = build_oracle_transaction(summary_date, index, new_oracle_data)
 
-      if Enum.empty?(new_oracle_data) do
-        Logger.debug("Oracle transaction skipped - no new data")
+    watcher_pid =
+      with {:empty, false} <- {:empty, Enum.empty?(new_oracle_data)},
+           {:trigger, true} <- {:trigger, trigger_node?(storage_nodes)},
+           {:exists, false} <- {:exists, DB.transaction_exists?(tx.address)} do
+        send_polling_transaction(tx)
+        nil
       else
-        send_polling_transaction(new_oracle_data, index, summary_date)
+        {:empty, true} ->
+          Logger.debug("Oracle transaction skipped - no new data")
+          nil
+
+        {:trigger, false} ->
+          {:ok, pid} =
+            DetectNodeResponsiveness.start_link(tx.address, fn count ->
+              if trigger_node?(storage_nodes, count) do
+                Logger.info("Oracle polling transaction ...attempt #{count}",
+                  transaction_address: Base.encode16(tx.address),
+                  transaction_type: :oracle
+                )
+
+                send_polling_transaction(tx)
```

Comment:
@imnik11 to avoid consensus not reached as the data might change, I think you should remake the data polling from the service providers